frontend/src/context/layers/admin_level_data.ts

```diff
@@ -58,14 +58,18 @@ export async function getAdminLevelDataLayerData({
       ? (LayerDefinitions[boundary as LayerKey] as BoundaryLayerProps)
       : getBoundaryLayerSingleton();
 
-  const adminBoundariesLayer = layerDataSelector(adminBoundaryLayer.id)(
+  let adminBoundariesLayer = layerDataSelector(adminBoundaryLayer.id)(
     getState(),
   ) as LayerData<BoundaryLayerProps> | undefined;
   // TEMP - add a 15s wait time to load admin boundaries which are very large
   // WARNING - This is a hack and should be replaced by a better handling of admin boundaries.
   // TODO - make sure we only run this once.
   if (!adminBoundariesLayer || !adminBoundariesLayer.data) {
     await new Promise(resolve => setTimeout(resolve, 15000));
+    // eslint-disable-next-line no-param-reassign
+    adminBoundariesLayer = layerDataSelector(adminBoundaryLayer.id)(
```

Comment:
@HarryMytilinaios I put this quick hack together to avoid the "boundary file not found" error but maybe there is a better way to do this? With a useEffect maybe or another way to re-trigger the layerDataSelector above?

Reply:
@ericboucher can you provide me some information when that error was happening, so that maybe we can find a better approach on that?

Reply:
the error was happening because when we first call the layerDataSelector, `adminBoundariesLayer` might be undefined because the boundary takes times to load.

Then we wait a bit for the boundary layer to load. But since we didn't reassign the variable, the code still thought that `adminBoundariesLayer` was undefined.

So I think what I did works ok, but maybe a useEffect might make more sense to have the component reload from the top and re-get the value for `adminBoundariesLayer`. 🤷